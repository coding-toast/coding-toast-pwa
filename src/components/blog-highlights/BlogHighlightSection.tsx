import * as React from 'react';
import { Button } from 'react-bootstrap';
import Link from 'next/link';
import SectionContainer, { BackgroundShade } from '../../layout/section-container/SectionContainer';
import BlogListDisplay from './BlogListDisplay';

const BlogHighlightSection: React.FC = () => {
  return (
    <SectionContainer backgroundShade={BackgroundShade.SECONDARY}>
      <h1>Catch Up On The Latest</h1>
      <BlogListDisplay />
      <Link href='/blogs'>
        <Button className='btn-accent-secondary' block>
          Checkout All Blog Posts
        </Button>
      </Link>
    </SectionContainer>
  );
};

export default BlogHighlightSection;
