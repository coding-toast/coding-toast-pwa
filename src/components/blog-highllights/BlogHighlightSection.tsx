import * as React from 'react'
import BlogHighlightSectionStyles from './BlogHighlightSection.module.css'
import { ExerptList } from './exerpt-list/ExerptList'

const BlogHighlightSection: React.FC<{}> = () => {
  return (
    <section className={BlogHighlightSectionStyles.container}>
      <h1>Catch Up On The Latest</h1>
      <ExerptList />
    </section>
  )
}

export default BlogHighlightSection
