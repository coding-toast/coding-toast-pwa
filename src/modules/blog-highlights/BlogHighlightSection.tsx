import * as React from 'react'
import { useStaticQuery, graphql } from 'gatsby'
import BlogHighlightSectionStyles from './BlogHighlightSection.module.scss'
import Container from '../../components/Container'
import ExcerptList from './excerpt-list/ExcerptList'

const BlogHighlightSection: React.FC = () => {
  const query = useStaticQuery(
    graphql`
      query {
        allMarkdownRemark(limit: 10, sort: { fields: frontmatter___publishDate, order: DESC }) {
          edges {
            node {
              id
              frontmatter {
                banner
                description
                title
                publishDate
              }
            }
          }
        }
      }
    `
  )

  return (
    <Container className={BlogHighlightSectionStyles.container}>
      <h1>Catch Up On The Latest</h1>
      <ExcerptList allMarkdownRemark={query} />
    </Container>
  )
}

export default BlogHighlightSection
