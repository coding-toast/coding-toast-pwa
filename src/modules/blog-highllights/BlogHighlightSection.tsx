import * as React from 'react'
import { useStaticQuery, graphql } from 'gatsby'
import BlogHighlightSectionStyles from './BlogHighlightSection.module.scss'
import Container from '../../components/Container'
import ExerptList from './exerpt-list/ExerptList'

const BlogHighlightSection: React.FC = () => {
  const query = useStaticQuery(
    graphql`
      query {
        allMarkdownRemark(limit: 10, sort: { fields: frontmatter___publishDate, order: DESC }) {
          edges {
            node {
              frontmatter {
                author
                banner
                description
                title
                publishDate
              }
            }
          }
        }
      }
    `
  )

  return (
    <Container className={BlogHighlightSectionStyles.container}>
      <h1>Catch Up On The Latest</h1>
      <ExerptList allMarkdownRemark={query} />
    </Container>
  )
}

export default BlogHighlightSection
