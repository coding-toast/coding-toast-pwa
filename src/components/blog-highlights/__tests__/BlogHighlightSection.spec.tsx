import * as React from 'react';
import { shallow } from 'enzyme';
import BlogHighlightSection from '../BlogHighlightSection';

describe('<BlogHighlightSection />', () => {
  it('It renders hopefully', () => {
    const foo = shallow(<BlogHighlightSection />);
    expect(foo).toMatchSnapshot();
  });
});
