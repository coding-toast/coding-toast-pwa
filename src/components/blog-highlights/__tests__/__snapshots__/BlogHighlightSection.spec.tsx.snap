// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlogHighlightSection /> It renders hopefully 1`] = `ShallowWrapper {}`;
